simpleclient/src/main/java/io/prometheus/client/Counter.java

```diff
@@ -108,6 +108,16 @@ protected Child newChild() {
    */
   public static class Child {
     private final DoubleAdder value = new DoubleAdder();
+    /**
+     *  Milliseconds since epoch. Optional value.
+     */
+    private Long timestamp;
+    /**
+     * Set the optional value for time stamp in epoch format.
+     */
+    public void setTimestamp(Long timestamp) {
```

Comment:
Direct instrumentation has no need for timestamps, these need to be removed.

Reply:
If I remove this, the _Unit Tests_ for `Counter` called `testCollectWithTimestamp` must be removed, because there will be no way for setting the _timestamp_ property (also happen the same for _Gauge_)). For me there is no problem so, are you agree to remove also the both _Unit Tests_ called `testCollectWithTimestamp`?

Reply:
Timestamps only work via custom collectors, so tests need to implement one of those.

Reply:
Ok, now I think I understand what you mean. But I'm afraid I have commited several changes. Tomorrow I will work on this.